griptape/tools/griptape_cloud_knowledge_base_client/tool.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+from urllib.parse import urljoin
+from schema import Schema, Literal
+from attr import define, field, Factory
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from griptape.artifacts import TextArtifact, ErrorArtifact
+
+
+@define
+class GriptapeCloudKnowledgeBaseClient(BaseTool):
+    """
+    Attributes:
+        description: LLM-friendly knowledge base description.
+        base_url: Base URL for the Griptape Cloud Knowledge Base API.
+        api_key: API key for Griptape Cloud.
+        headers: Headers for the Griptape Cloud Knowledge Base API.
+        knowledge_base_id: ID of the Griptape Cloud Knowledge Base.
+    """
+
+    description: str = field(kw_only=True)
```

Comment:
Can we pull description dynamically with an optional override (what I have in the POC)? Could be a very nice UX feature for true end-to-end.

Reply:
Added